### ISSUE TITLE
fix chef_version check from >=12.5 to >=12.6 or >=12.7

### ISSUE THREAD

**Issue by akihiro** [chef-cookbooks/docker#863]
### Cookbook version
2.15.6

### Chef-client version
12.5.1 or 13.1.31

### Platform Details
CentOS Linux release 7.3.1611 (Core) on VirtualBox5

### Scenario:
docker cookbook is called by chef 12.5.1 is failed. but by chef 13.1.31 is succeeded.

README.md and metadata.rb is support 12.5.x or higher.

In this thred https://discourse.chef.io/t/gem-in-cookbook-libraries-what-is-the-right-way/9014 , 
> We added official support for this in 12.6 or 12.7 (I forget which). Add gem 'docker-api' , '~> 1.28' to your metadata.rb just like you would depend on a cookbook.

### Steps to Reproduce:
```console
$ echo "cookbook 'docker', '~> 2.15.6'" >> Berksfile
$ sudo chef-client -z -c ./zero.rb
```

### Expected Result:
```console
$ sudo chef-client -z -c ./zero.rb
Starting Chef Client, version 13.1.31
resolving cookbooks for run list: ["foobar::docker"]
Synchronizing Cookbooks:
  - foobar(0.0.1)
  - docker (2.15.6)
Installing Cookbook Gems:
Compiling Cookbooks...
Converging 1 resources
Recipe: foobar::docker
  * docker_installation_script[default] action create
    * yum_package[curl] action install (up to date)
    * execute[install docker] action run
      - execute curl -sSL https://get.docker.com/ | sh


Running handlers:
Running handlers complete
Chef Client finished, 2/3 resources updated in 34 seconds
```

### Actual Result:
```console
$ chef-client -W -z -c ./zero.rb
Starting Chef Client, version 12.5.1
resolving cookbooks for run list: ["foobar::docker"]
Synchronizing Cookbooks:
  - foobar (0.0.1)
  - docker (2.15.6)
Compiling Cookbooks...

================================================================================
Recipe Compile Error in /local-mode-cache/cache/cookbooks/docker/libraries/docker_container.rb
================================================================================

LoadError
---------
cannot load such file -- docker

Cookbook Trace:
---------------
  /local-mode-cache/cache/cookbooks/docker/libraries/docker_container.rb:3:in `<class:DockerContainer>'
  /local-mode-cache/cache/cookbooks/docker/libraries/docker_container.rb:2:in `<module:DockerCookbook>'
  /local-mode-cache/cache/cookbooks/docker/libraries/docker_container.rb:1:in `<top (required)>'

Relevant File Content:
----------------------
/local-mode-cache/cache/cookbooks/docker/libraries/docker_container.rb:

  1:  module DockerCookbook
  2:    class DockerContainer < DockerBase
  3>>     require 'docker'
  4:      require 'shellwords'
  5:      require_relative 'helpers_container'
  6:
  7:      include DockerHelpers::Container
  8:
  9:      resource_name :docker_container
 10:
 11:      ###########################################################
 12:      # In Chef 12.5 and later, we no longer have to use separate


Running handlers:
[2017-06-01T18:35:36+09:00] ERROR: Running exception handlers
Running handlers complete
[2017-06-01T18:35:36+09:00] ERROR: Exception handlers complete
Chef Client failed. 0 resources would have been updated
[2017-06-01T18:35:36+09:00] FATAL: Stacktrace dumped to /local-mode-cache/cache/chef-stacktrace.out
[2017-06-01T18:35:36+09:00] ERROR: cannot load such file -- docker
[2017-06-01T18:35:38+09:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

**Comment by chasebolt** [comment]
resolved in release 2.15.10 (dbf134773d3f6ea2c575c2c7e114e05e3e0cff11)

**Comment by chasebolt** [comment]
had to revert this in 2.15.12. gem in the metadata doesnt not include chef's builtin gems. this requires the docker-api gem to compile the json gem, creating a dependency on build tools.
